### PR TITLE
milestone 0 

### PR DESCRIPTION
**Pull request by championswimmer**
* use cardview 
* show more details in sessions, speakers and tracks
* move from ngRoute to ui.router (will help in future, as it's more flexible to set routes and states)
* save fetched data in sessionStorage

### CODE REVIEW

**Comment by championswimmer** [comment]
@creativepsyco @leto 
Please review and merge

File: app/components/sessions/SessionsController.js [hunk 1]
```diff
@@ -2,24 +2,40 @@
...
 }]);
 
 sessionsModule.controller('SessionsController',
-    ['$rootScope', 'ApiJsonFactory', function($rootScope, ApiJsonFactory) {
+    ['$sessionStorage', '$rootScope', 'ApiJsonFactory', function($sessionStorage, $rootScope, ApiJsonFactory) {
         var sc = this;
-        sc.Sessions = [];
+        sc.$storage = $sessionStorage;
+        if (typeof(sc.$storage.sessions) == 'undefined' || sc.$storage.sessions == null)
+        {
```
Comment (creativepsyco) [inline app/components/sessions/SessionsController.js]: style knit.

File: app/components/tracks/TracksController.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        templateUrl: 'app/components/tracks/tracks.html',
+        controller: 'TracksController'
+    })
+}]);
+
+tracksModule.controller('TracksController', 
+	['$sessionStorage', '$rootScope', 'ApiJsonFactory', function($sessionStorage, $rootScope, ApiJsonFactory) {
+		var sc = this;
+        sc.$storage = $sessionStorage;
+        if (typeof(sc.$storage.tracks) == 'undefined' || sc.$storage.tracks == null)
```
Comment (creativepsyco) [inline app/components/tracks/TracksController.js]: should check for `null` first.

File: app/libs/angular-storage.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
+(function(angular, factory) {
```
Comment (creativepsyco) [inline app/libs/angular-storage.js]: The formatting does not match the rest of the files in the project. You should provide a License if this is Open Source code or if this is your code, then format it accordingly.

**Comment by creativepsyco** [comment]
minor comments.

**Comment by mananwason** [comment]
@championswimmer please try to squash commits like @creativepsyco has told you earlier as well

**Comment by championswimmer** [comment]
I had squashed the ones that were related.

File: app/components/sessions/SessionsController.js (hunk 1 above)
Comment (championswimmer) [inline app/components/sessions/SessionsController.js]: didn't get what you meant by that

**Pull request merged**